ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,76 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/http-proxy/filters"
+	"github.com/hashicorp/golang-lru"
+)
+
+var log = golog.LoggerFor("ratelimiter")
+
+type ratelimiter struct {
+	hostPeriods          map[string]time.Duration
+	hostAccessesByClient *lru.Cache
+	mx                   sync.Mutex
+}
+
+// New creates a new rate limiting filter that only allows access to the hosts
```

Comment:
I added a package-level comment.  Did you see the existing explanation on the New() function too?